### PR TITLE
[CPU] Enable dynamic quantization by default

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *Enable dynamic quantization by default*
 - *...*

### Tickets:
 - *[134931](https://jira.devtools.intel.com/browse/CVS-134931)*

### CODE REVIEW

File: src/plugins/intel_cpu/src/config.h [hunk 1]
```diff
@@ -50,8 +50,8 @@ struct Config {
     std::string dumpToDot = {};
     std::string device_id = {};
     float fcSparseWeiDecompressionRate = 1.0f;
-    uint64_t fcDynamicQuantizationGroupSize = 0;
-    ov::element::Type kvCachePrecision = ov::element::f16;
+    uint64_t fcDynamicQuantizationGroupSize = 32;
+    ov::element::Type kvCachePrecision = ov::element::u8;
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/plugins/intel_cpu/src/config.h:54]: can we cover such behavior in tests for ACCURACY mode?

File: src/plugins/intel_cpu/src/config.cpp [hunk 2]
```diff
@@ -348,6 +348,15 @@ void Config::readProperties(const ov::AnyMap& prop, const ModelType modelType) {
...
             try {
                 executionMode = val.as<ov::hint::ExecutionMode>();
+                if (executionMode == ov::hint::ExecutionMode::ACCURACY) {
+                    // disable dynamic quantization and kv quantization for best accuracy
+                    fcDynamicQuantizationGroupSize = 0;
+                    kvCachePrecision = ov::element::f32;
+                } else {
+                    // restore to default
+                    fcDynamicQuantizationGroupSize = 32;
+                    kvCachePrecision = ov::element::u8;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/config.cpp:358]: as I know, FP32 is also used as inference precision of ACCURACY mode is used even on Xeon with BF16.
Is it handled in a different place?

File: src/plugins/intel_cpu/src/config.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/config.h:54]: Added.

File: src/plugins/intel_cpu/src/config.cpp (hunk 2 above)
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/config.cpp]: Move changes to the location of inference precision of ACCURACY: https://github.com/openvinotoolkit/openvino/blob/5d2da3bef5de5784e9a37ab85ff43aab05a9ef79/src/plugins/intel_cpu/src/config.cpp#L381-L404

**Review by maxnick — APPROVED** [review]

**Comment by maxnick** [comment]
@ilya-lavrenov , do you have any further comments or suggestions?

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**